src/test/scala-2/chiselTests/experimental/hierarchy/InstantiateSpec.scala

```diff
@@ -190,99 +191,106 @@ class InstantiateSpec extends AnyFunSpec with Matchers with Utils {
 
   import InstantiateSpec._
 
+  private val ExtractModuleName = """.*module\s+(\w+)\s*:.*""".r
+
+  private def elaborateAndGetModuleNames(gen: => RawModule): Seq[String] = {
+    val chirrtl = emitCHIRRTL(gen)
+    chirrtl.split("\n").collect { case ExtractModuleName(name) => name }.toSeq
+  }
```

Comment:
I'm okay with this, but would greatly prefer directly using `FileCheck` in the tests. I realize that this is a heavier lift given how this test is written.

Reply:
Fortunately, it was _relatively_ easy using AI so I went ahead and updated it to FileCheck.